tests/run-pass/concurrency/pthread_condattr_init.rs

```diff
@@ -0,0 +1,20 @@
+// ignore-windows: No libc on Windows
+// compile-flags: -Zmiri-check-number-validity
+
+#![feature(rustc_private)]
+
+/// Test that pthread_condattr_destroy doesn't trigger a number validity error.
+extern crate libc;
```

Comment:
Instead of adding a new test, could you add this to `libc_pthread_cond.rs`?

Also, please add the `let _ = std::sync::Condvar::new();` from your bug report to the `sync.rs` test. How is it even possible that that test passed without you fix...?

Reply:
That test uses *zeroed* memory, not uninit. Maybe written before MaybeUninit? In any case, I'll move it over to use MaybeUninit::uninit

Reply:
> Edit: It also doesn't run with the flag for number validity, so nothing really cares that it's uninit without that.

D'oh, of course. We should add that flag to `sync.rs`. I confirmed that that is indeed enough to trigger the problem.